scripts/run-e2e-test.sh

```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+# Run end-to-end tests
+
+export PICCOLO_CONF="e2e.piccolo_conf"
+
+python -m pytest e2e 
```

Comment:
One of the main differences between Playwright and Cypress is Playwright doesn't emit images / videos by default. We need to pass in the `--screenshot=on`, and `--video=on` flags.

https://playwright.dev/python/docs/test-runners#cli-arguments

It would be nice to upload these as build artefacts (not sure if screenshots/videos/both, and whether just on failure). It will help us see why tests fail.

The Cypress GitHub action had some code we can reuse:

```
      - name: "Upload Cypress Screenshots and Videos"
        uses: actions/upload-artifact@v3
        with:
          name: cypress-artifacts
          path: |
            /home/runner/work/piccolo_admin/piccolo_admin/admin_ui/cypress/videos/
            /home/runner/work/piccolo_admin/piccolo_admin/admin_ui/cypress/screenshots/
        if: always()
```

Reply:
Ok, I'll try adding that and see how it works. Now [act](https://github.com/nektos/act) comes very handy because I can try everything locally, not by pushing commits to Github.

Reply:
I tried `--screenshot=only-on-failure` and `--screenshot=on` . I made `screenshots` folder inside `e2e` and use
```yaml
- name: "Upload Playwright Screenshots"
  uses: actions/upload-artifact@v3
  with:
    name: playwright-artifacts
    path: |
      /home/runner/work/piccolo_admin/piccolo_admin/e2e/screenshots/
  if: always()
```
and 
```yaml
- uses: actions/upload-artifact@v2
  if: always()
  with:
    name: screenshots
    path: screenshots/
    retention-days: 5
```
but result is always same. I get this warning `::warning::No files were found with the provided path: /home/runner/work/piccolo_admin/piccolo_admin/e2e/screenshots/. No artifacts will be uploaded.` and `screenshots` folder is empty.
In the end I put `--screenshot=only-on-failure`. Do you have any idea how to save screenshots?

Reply:
I think the folder name might be `test-results' instead of `screenshots`.

For some other Playwright tests I have, the path looks like this:

```
test-results/integration-test-signup-py-test-signup-chromium/test-failed-1.png
```

I think it' possible to customise the path using the `--output` flag.

Reply:
Sorry but I was unable to set the `test-results`. I don't know where the problem is and I always get `::warning::No files were found with the provided path` on `act` tool. I tried to put `test-results` in `e2e` and in `root`. I tried `--output` flag but nothing works. GH actions says that artifact is successfully uploaded but files size is 0. If you can fix it somehow, that would be great. Apart from that (all tests and GH actions) everything else works fine.